snap/types.go

```diff
@@ -38,6 +38,17 @@ const (
 	TypeOS Type = "os"
 )
 
+// This is the sort order from least important to most important for
+// types. On e.g. firstboot this will be used to order the snaps this
+// way.
+var typeOrder = map[Type]int{
+	TypeApp:    40,
+	TypeBase:   30,
+	TypeGadget: 20,
```

Comment:
gadget will need  a base as well, so either all bases should come before gadget, or we will always need to tweak the   resulting order to put the gadget's base before the gadget